### PR TITLE
Make some small improvements

### PR DESCRIPTION
**Pull request by Moosems**

### CODE REVIEW

**Comment by Moosems** [comment]
I wonder if there’s a way to create a base page and then just modify certain sections similar to how flask does it.

**Comment by not-nef** [comment]
i am currently in the middle of making the dynamic member list, can i wait with this until thats done? there will probably be a lot of merge conflicts

**Comment by Moosems** [comment]
> i am currently in the middle of making the dynamic member list, can i wait with this until thats done? there will probably be a lot of merge conflicts

Just make it into a js script that puts it into in a div with the id `members`. I can integrate that easily.

**Comment by not-nef** [comment]
im already pretty much done ill just have to do some testing, but i exceeded my api rate limit and have to wait a bit

**Comment by Moosems** [comment]
Screen Shot 2023-09-12 at 12 23 04 PM
All you need to do is make tons of these divs and then paste it into the document 😉.

**Comment by not-nef** [comment]
i think you get what i want to do when you see it

**Comment by Moosems** [comment]
If it's just a JS modification you can add to landing.js and it should work fine 😄.

**Comment by not-nef** [comment]
https://github.com/Futura-Py/futura-py.github.io/blob/main/scripts.js

you see what i did?

**Comment by not-nef** [comment]
it dynamically generates the divs which is cool

**Comment by not-nef** [comment]
what do you think?

**Comment by Moosems** [comment]
I like it.

**Comment by not-nef** [comment]
i do want to keep the modern feel for this so i want to make some adjustments once you are done

**Comment by Moosems** [comment]
Have my changes not been modern looking?

**Comment by not-nef** [comment]
i think the boxes and the text shadows (especially on the paragraphs) dont look that great, although i get why you made the boxes, but well see

**Comment by Moosems** [comment]
I thought the shadows looked nice 🙁. The boxes were made cause of contrast issues

**Comment by not-nef** [comment]
i was aware of the contrast issues and was just gonna make the links another color, but the shadows look pretty confusing on the paragraphs, they can stay on the other elements but i dont like them on the paragraphs

**Comment by not-nef** [comment]
let me know when youre finished with this pr so that i can make some adjustions

**Comment by Moosems** [comment]
Ok, you're free to mess around with it for a bit.

**Comment by not-nef** [comment]
im done with coding for today, ill look at it tomorrow

File: index.html
Comment (not-nef) [inline index.html]: isnt the second one kind of unnecessary?

**Comment by im-coder-lg** [comment]
Can we soon change to Astro? I have tried it, and it is robust and very fast. I think we can integrate some _cool_ features with it. Of course, someone gotta know CSS.

**Comment by Moosems** [comment]
Y'all really love making simple things complicated, huh?

**Comment by im-coder-lg** [comment]
I don't know, it just happens in my case. All the time. But then I would have clarity, and I don't know a deuce about how that happens too. But I have used Astro, and it is kinda simple. The main files use JSX if I am not wrong, and it actually allows HTML in the code, if laid properly. I use it on my Astrosite. It is the literal repo name btw.

**Comment by not-nef** [comment]
> Can we soon change to Astro? I have tried it, and it is robust and very fast. I think we can integrate some _cool_ features with it. Of course, someone gotta know CSS. 

i see no need for it

**Comment by im-coder-lg** [comment]
Well, I do. My idea isn't fully formed yet, but the way I see it, if we make the website out of Astro, it would be loads faster. Plus, there are many plugins for Astro out there, we can use them. HTML and JS are used in the same file, mind you. Same with Markdown, we can use Markdown and JavaScript. So, any cool embeds that we have to script can be done immediately along with running HTML side-by-side. I will try this idea later, it is prone to work.

**Comment by not-nef** [comment]
well you can try it on a different branch and ill look at it

**Comment by not-nef** [comment]
i need some help: i cant get the "about us" and "projects" to align left. i have been messing around with the css for about half an hour and couldnt get it to work. can anyone of you help me?

**Comment by Moosems** [comment]
> Well, I do. My idea isn't fully formed yet, but the way I see it, if we make the website out of Astro, it would be loads faster. Plus, there are many plugins for Astro out there, we can use them. HTML and JS are used in the same file, mind you. Same with Markdown, we can use Markdown and JavaScript. So, any cool embeds that we have to script can be done immediately along with running HTML side-by-side. I will try this idea later, it is prone to work.

Please tell me how adding extra bloat would make it load faster then barebones html 😄.

**Comment by not-nef** [comment]
and html is just markdown on crack so why would we need it

**Comment by sumeshir26** [comment]
> Well, I do. My idea isn't fully formed yet, but the way I see it, if we make the website out of Astro, it would be loads faster. Plus, there are many plugins for Astro out there, we can use them. HTML and JS are used in the same file, mind you. Same with Markdown, we can use Markdown and JavaScript. So, any cool embeds that we have to script can be done immediately along with running HTML side-by-side. I will try this idea later, it is prone to work.

Why do we need to over-engineer such a simple website? I dont see any speed problems now, and the codebase isnt that hard to work with.

**Comment by sumeshir26** [comment]
> i need some help: i cant get the "about us" and "projects" to align left. i have been messing around with the css for about half an hour and couldnt get it to work. can anyone of you help me?

Maybe using `<div allign="left">`?

**Comment by not-nef** [comment]
well that would be embarrassing if i missed that, i will try it today

**Comment by Moosems** [comment]
> Why do we need to over-engineer such a simple website? I dont see any speed problems now, and the codebase isnt that hard to work with.

When I say I dislike a suggestion, this is how k feel 95% of the time by the way. I don't dislike y'all's suggestions but y'all really have a knack for over complicating the applications.

**Comment by im-coder-lg** [comment]
Well, I am not trying to overcomplicate stuff. I think you are thinking we get a template. We are absolutely NOT doing that. Instead, we could just use the present code in the Astro files. Oh, I almost forgot. Astro files have a `.astro` extension, and by using a plugin, we might be able to transcribe the HTML/Markdown/JSX into Astro files on build.

And Astro doesn't complicate stuff. Rather, I believe the code built will be just HTML and JS that is very rapid. Astro wouldn't have to play a part in the build files, it plays a part in the source code. You can see [my most recent website's source code](https://GitHub.com/im-coder-lg/Astrosite) - it is rapid even with that many files. 

Yeah, still a work in progress. But that is with a template. Imagine the speeds we could get if we create our own theme and add our type of icons? I'm speaking Font Awesome(I heard there was a free edition?), MUI, and our custom SVG icons if we want. That is where JS could help. But this is my pitch, you guys should decide.

**Comment by not-nef** [comment]
> > i need some help: i cant get the "about us" and "projects" to align left. i have been messing around with the css for about half an hour and couldnt get it to work. can anyone of you help me?
> 
> Maybe using `<div allign="left">`?

didnt work. @Moosems you have any idea

**Comment by Moosems** [comment]
Haven't run it yet, will give some tries in an hour.

**Comment by im-coder-lg** [comment]
https://github.com/Futura-Py/futura-py.github.io/issues/2#issuecomment-1719326873

And yeah, I'm testing Astro for this website. It _will_ work, and I know it. On my fork, on another branch in it.

**Comment by not-nef** [comment]
ya boy is 15 now

**Comment by sumeshir26** [comment]
> ya boy is 15 now

Happy Birthday!

**Comment by im-coder-lg** [comment]
> ya boy is 15 now

Happy birthday bro!

**Comment by Moosems** [comment]
Happy birthday my dude!

**Comment by im-coder-lg** [comment]
Oh, and a bit of a side note, which ofc can be checked later: https://github.com/Futura-Py/futura-py.github.io/pull/2#issuecomment-1719326873

I'm testing Astro as well as recreating the website we are using, with CSS and JS, plus some interesting goodies I found, like a card view that seems similar to the one used in Next.js, and more. I'll test and show my results later.
It is still snappy.

**Comment by Moosems** [comment]
@not-nef I'm fine with the logo being to the side but can we have the different items centered and keep the sticky header? I worked really hard getting the sticky header in particular and want to keep it. To be honest, could we just keep it how it was (I'm fine with the color change)?

**Comment by not-nef** [comment]
thanks yall for the birthday wishes! i just liked the nav bar being like over the banner your know? we can change it back if you like. i do want the items to align left you know. it would be awesome if you could get it to work. but if you dont thats fine. well find a solution

**Comment by Moosems** [comment]
I'm updating the sticky header now, please don't touch it afterwards.

**Comment by Moosems** [comment]
I recommend taking a look at apple.com and their navbar.

**Comment by not-nef** [comment]
> I'm updating the sticky header now, please don't touch it afterwards.

alright, tell me when i can lay my hands on it again

**Comment by im-coder-lg** [comment]
Uhh, guys? The astrosite's going well... I'll try to furnish it to make it look like the present version of our site, then add some more stuff once this PR is merged. I used just HTML if I think about it. Yeah, CSS and JS are included.

**Comment by Moosems** [comment]
@im-coder-lg You're keeping my navbar right?

**Comment by im-coder-lg** [comment]
Well, hell yeah. You guys finish this PR.
@Moosems see, when I was creating the website with Astro, I realised that the Astro files for site generation use HTML, JS and TS.

There is the frontmatter-like part where you put your JS code, perhaps imports to use later, then you use HTML, and wherever needed, use JS or TS. The way I have done is taking most of the code from here, using it there, and slowly but surely refactoring it so that it not only becomes a 1:1 copy of this, but also can use some neat features like Astro Cards, which is basically the Card UI in HTML(tell me if I am warm) and more. There is an unseen advantage to this: adding JS scripts are much more easier, less painstaking, and moreover, we can add plugins like buttons, icons, symbols and more. I have tested this out on multiple websites on my profile. There are so many, I seem to have lost count(my most memorable one was my Jekyll blog, until I deleted that. It had over 200 commits.)
I saw my Next.js course blog, and I saw this: https://dub.sh/EUAhImA

Had to shorten it, so that I can now refer to it anywhere. Anyway, it's my website, where I learnt basic Next.js. Anyway, what I want you guys to know is that although the website might be statically generated, I think the API renders along with the server, making it a tad slow. But hey, :shrug: it's your decision now, I gave the idea. What do you guys think about this?

**Comment by not-nef** [comment]
should we maybe get rid of the banner for the base and only keep it for that main page? it takes up like half the screen

**Comment by not-nef** [comment]
also can i mess around with it a bit?

**Comment by Moosems** [comment]
Yes, let's keep the navbar as it is for now though.

**Comment by im-coder-lg** [comment]
Yeah. You guys code with HTML, I'll slowly add it to Astro.

Btw I'm 15.

Edit: 15 now, for clarity. Seeing many September-borns such as me, @not-nef and a couple of people from my school. Very funny to see that that bunch in my school share my birthday. Lol

**Comment by Moosems** [comment]
Why is everyone sharing their age 😄?

**Comment by im-coder-lg** [comment]
Hmm, we got our birthdays? I saw two other people having their birthday on my birthday. Well, we could call this "month of most known birthdays" or something lol

**Comment by not-nef** [comment]
> Why is everyone sharing their age 😄?

We already knew ours from back in the timerx days so its no secret anymore. also im not ashamed for my age

**Comment by not-nef** [comment]
> Yes, let's keep the navbar as it is for now though.

got it

**Comment by sumeshir26** [comment]
> Yeah. You guys code with HTML, I'll slowly add it to Astro.

> Btw I'm 15.

> Edit: 15 now, for clarity. Seeing many September-borns such as me, @not-nef and a couple of people from my school. Very funny to see that that bunch in my school share my birthday. Lol

I'm august born lol

**Comment by not-nef** [comment]
the company sent me a new cpu (thank god). i have now 100% switched to linux (i even tried installing arch but miserably failed getting budgie to work)

**Comment by im-coder-lg** [comment]
> the company sent me a new cpu (thank god). i have now 100% switched to linux (i even tried installing arch but miserably failed getting budgie to work)

Try blendOS. It actually allows you to use Ubuntu and Fedora packages. I haven't found a need to distro-hop, now that I am careful about wrecking the OS, but I recommend that. It is really good at what it does.

**Comment by im-coder-lg** [comment]
> > Yeah. You guys code with HTML, I'll slowly add it to Astro.
> 
> > Btw I'm 15.
> 
> > Edit: 15 now, for clarity. Seeing many September-borns such as me, @not-nef and a couple of people from my school. Very funny to see that that bunch in my school share my birthday. Lol
> 
> I'm august born lol
> 

I tried quoting, but edited it, on my Android app. Well, anyway, best guy I ever know, one of the best August-borns, you are.

Don't know why I spoke like Yoda. Whatevs.

**Comment by sumeshir26** [comment]
> > > Yeah. You guys code with HTML, I'll slowly add it to Astro.
> > 
> > 
> > > Btw I'm 15.
> > 
> > 
> > > Edit: 15 now, for clarity. Seeing many September-borns such as me, @not-nef and a couple of people from my school. Very funny to see that that bunch in my school share my birthday. Lol
> > 
> > 
> > I'm august born lol
> 
> I tried quoting, but edited it, on my Android app. Well, anyway, best guy I ever know, one of the best August-borns, you are.
> 
> Don't know why I spoke like Yoda. Whatevs.

yep totally getting yoda vibes
thx btw lol

**Comment by Moosems** [comment]
Why not merged this PR is, I know not.

**Comment by not-nef** [comment]
Screenshot from 2023-09-25 17-23-24

**Comment by Moosems** [comment]
> Screenshot from 2023-09-25 17-23-24

I can fix all this if you want... The Futura logo is gonna look real small though.

**Comment by not-nef** [comment]
i think a navbar logo shouldnt be that big tbh (see github navbar or apple for example thats how i think it should be)

**Comment by im-coder-lg** [comment]
> Why not merged this PR is, I know not.

What the deuce did I just start?!

> > Screenshot from 2023-09-25 17-23-24

> I can fix all this if you want... The Futura logo is gonna look real small though.

That is how it always is with navbars. If you guys finish this, I can add it to Astro. Then we will add that codebase here, upon which we can use the Astro Community components to make the website better.

**Comment by not-nef** [comment]
sorry but i still don't see the benefit of using astro. you can do your thing on a seperate branch, but until then we will continue development as usual

**Comment by not-nef** [comment]
also did we settle on removing the banner from the page template?

**Comment by Moosems** [comment]
I'll update the navbar, Astro is a separate branch until proven worthy of pushing to main, and banner is only on homepage. Matters settled :).

**Comment by im-coder-lg** [comment]
Astro is _not_ a branch here, at least not here for now. It is on my local system, and I want to host it somewhere accessible to actually run the website off of Netlify or something. Now that I think of it, I might as well add a CDN to edit pages on the go. Maybe as part of a small test. And Astro isn't a waste of time. It is actually well-structured and understandable for people who try it, so please do so. It uses HTML but with perks and batteries included.

**Comment by im-coder-lg** [comment]
Alright everyone, the code's [here](https://dub.sh/3yfHx03), with a task list.

**Comment by not-nef** [comment]
will check when im home, but this looks good!

**Comment by Moosems** [comment]
Loaded it with mobile preview and it looks pretty good there too :).

**Comment by not-nef** [comment]
i really wanted to approve this. i really wanted to. but i cant:
Screenshot from 2023-09-28 20-05-17

**Comment by Moosems** [comment]
I'll fix that tonight (hopefully).

**Comment by Moosems** [comment]
Can I push this to main y'all?

**Comment by sumeshir26** [comment]
> Can I push this to main y'all?

Sure, go ahead

**Comment by not-nef** [comment]
can i take a look at it one last time? ill be home in a few hours. (i think i missed your last commit somehow.) ive been wanting to get back into programming anyway.

**Review by not-nef — APPROVED** [review]
me happy

**Pull request merged**
